src/compute-client/src/metrics.rs

```diff
@@ -704,3 +810,34 @@ impl<M> PeekMetrics<M> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[mz_ore::test]
+    fn minmax() {
+        let mut minmax = MinMax::new(5);
+
+        assert_eq!(minmax.get(), None);
+
+        let mut push_and_check = |x, expected| {
+            minmax.add_sample(x);
+            let actual = minmax.get().map(|(min, max)| (*min, *max));
+            assert_eq!(actual, Some(expected), "{minmax:?}");
+        };
+
+        push_and_check(5, (5, 5));
+        push_and_check(1, (1, 5));
+        push_and_check(10, (1, 10));
+        push_and_check(2, (1, 10));
+        push_and_check(9, (1, 10));
+        push_and_check(3, (1, 10));
+        push_and_check(8, (2, 10));
+        push_and_check(5, (2, 9));
+        push_and_check(5, (3, 9));
+        push_and_check(5, (3, 8));
+        push_and_check(5, (5, 8));
+        push_and_check(5, (5, 5));
+    }
+}
```

Comment:
This feels general purpose enough that I think it should move to the `ore::stats` module. No need to do taht in this PR though.